numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -1175,6 +1175,41 @@ dtypemeta_wrap_legacy_descriptor(
         Py_DECREF(dtype_class);
         return NULL;
     }
+
+    /* Set the text signature for inspect.signature() */
+    const char *base_doc = ((PyTypeObject *)dtype_class)->tp_doc;
```

Comment:
Might as well move the base doc from above here, since we set `tp_doc` later anyway I think.